### PR TITLE
chore: libwaku reduce repetitive code by adding a template handling resp returns

### PR DESCRIPTION
**Pull request by Ivansete-status**
## Description
This is inspired by SP's suggestion: https://github.com/waku-org/nwaku/pull/3026#discussion_r1750892090

### CODE REVIEW

File: library/libwaku.nim [hunk 1]
```diff
@@ -52,6 +52,21 @@ template foreignThreadGc(body: untyped) =
...
 
+template handleRes[T, E](res: Result[T, E], callback: WakuCallBack, userData: pointer) =
+  ## Manages sendRequestToWakuThread responses in a generic way
+  if res.isErr():
+    foreignThreadGc:
+      let msg = "libwaku error: " & $res.error
+      callback(RET_ERR, unsafeAddr msg[0], cast[csize_t](len(msg)), userData)
+      return RET_ERR
+
+  when T is string:
```
Comment (NagyZoltanPeter) [inline library/libwaku.nim:63]: It seems me suspicious, but maybe overlooking sthg.
So what if T is not a string?
I see code that the template replaces which use stringizing the value of the return type, whatever it is.
I think it is not exactly the same behavior.
Also if it is still ok for now, I would add an else case to result in compile error thou shall not hide a problem of misuse.

**Review by NagyZoltanPeter — CHANGES_REQUESTED** [review]
All in all it is a super step I do appreciate a lot.
Please have a look at my concern in handling OK case inside the template.

File: library/libwaku.nim (hunk 1 above)
Comment (SionoiS) [reply library/libwaku.nim:63]: In Rust trait bounds are used to restrict types when using generics any such thing in Nim?

**Review by SionoiS — APPROVED** [review]

File: library/libwaku.nim (hunk 1 above)
Comment (arnetheduck) [reply library/libwaku.nim:63]: `template handleRes[T: string, ...`

**Review by NagyZoltanPeter — APPROVED** [review]
Thanks for it!!!

File: library/libwaku.nim (hunk 1 above)
Comment (Ivansete-status) [reply library/libwaku.nim:65]: Thank you for the comments guys!
Applied it in the following commits:

[libwaku: narrow down the possible types used by handleRes](https://github.com/waku-org/nwaku/pull/3032/commits/bfb241a2c32b3dbc00edb2bf1676773339f0d6a1)

[libwaku: limit the possible types accepted by handleRes](https://github.com/waku-org/nwaku/pull/3032/commits/fa06e3421e0ae898017833d879a6f627803759be)

**Pull request merged**